### ISSUE TITLE
关于loss函数

### ISSUE THREAD

**Issue by aaashinshin** [FlagOpen/FlagEmbedding#349]
作者您好，我看您的微调时使用的交叉熵损失
if self.training:
    if self.negatives_cross_device:
        q_reps = self._dist_gather_tensor(q_reps)
        p_reps = self._dist_gather_tensor(p_reps)

    scores = self.compute_similarity(q_reps, p_reps)
    scores = scores / self.temperature
    scores = scores.view(q_reps.size(0), -1)

    target = torch.arange(scores.size(0), device=scores.device, dtype=torch.long)
    target = target * (p_reps.size(0) // q_reps.size(0))
    loss = self.compute_loss(scores, target)
    
def compute_loss(self, scores, target):
    return self.cross_entropy(scores, target)
    
self.cross_entropy = nn.CrossEntropyLoss(reduction='mean')
    
想请问一下这里能改成Triplet Loss吗？如果能，和交叉熵比有何优劣呢？劳烦您解惑

**Comment by staoxiao** [comment]
可以的。交叉熵使用起来更方便，一次性能使用多个负样本，Triplet Loss一次只有一个负样本。

**Comment by aaashinshin** [comment]
> 可以的。交叉熵使用起来更方便，一次性能使用多个负样本，Triplet Loss一次只有一个负样本。

明白了，感谢您的回复。
